lib/manageiq/providers/openstack/legacy/events/openstack_stf_event_monitor.rb

```diff
@@ -66,24 +65,20 @@ def stop
 
   def each_batch
     while @collecting_events && @qdr_receiver.running > 0
-      @events_mutex.synchronize do
-        converted_events = @events.map do |raw_event|
-          unserialized_event = unserialize_event(raw_event)
-          converted_event = OpenstackStfEventConverter.new(unserialized_event)
-          $log.debug("Processing a new OpenStack STF Event: #{unserialized_event.inspect}") if $log
-          openstack_event(nil, converted_event.metadata, converted_event.payload)
-        end
-
-        filtered_events = filter_event_types(converted_events)
-
-        $log.info("MIQ(#{self.class.name}) STF Yielding #{filtered_events.size} events to"\
-        " event_catcher: #{filtered_events.map { |e| e }}") if $log
-
-        yield filtered_events
-
-        $log.info("MIQ(#{self.class.name}) Clearing events") if $log && @events.any? && filtered_events.any?
-        @events.clear
+      captured_events = []
+      while !@events.empty? && captured_events.length < EVENTS_BATCH_MAX_SIZE
+        captured_events << @events.pop
+      end
+      converted_events = captured_events.map do |raw_event|
+        unserialized_event = unserialize_event(raw_event)
+        converted_event = OpenstackStfEventConverter.new(unserialized_event)
+        $log.debug("Processing a new OpenStack STF Event: #{unserialized_event.inspect}") if $log
+        openstack_event(nil, converted_event.metadata, converted_event.payload)
       end
+      filtered_events = filter_event_types(converted_events)
+      $log.info("MIQ(#{self.class.name}) STF Yielding #{filtered_events.size} events to"\
+      " event_catcher: #{filtered_events.map { |e| e }}") if $log
```

Comment:
@aufi so this will log an empty array every 5 seconds?  Also didn't notice this before but what is `filtered_events.map { |e| e }` doing?